app/controllers/items/new.js

```diff
@@ -69,6 +70,23 @@ export default GoodcityController.extend(
       return this.get("i18n").t(str);
     },
 
+    allAvailablePrinters: Ember.computed(function() {
+      return this.get("printerService").allAvailablePrinters();
+    }),
+
+    selectedPrinterDisplay: Ember.computed("inventoryNumber", {
+      get(key) {
+        let userPrinter = this.get("session").userDefaultPrinter();
+        if (!userPrinter) {
+          return;
+        }
+        return { name: userPrinter.get("name"), id: userPrinter.id };
+      },
+      set(key, value) {
```

Comment:
The usage of a setter (which doesn't seem to set anything anyways) in combination with a computed property which reacts to a property `inventoryNumber` is definitely a code smell.

The only way it would make sense is if the `set` method actually modifies the `inventoryNumber` which in turn triggers the property to update.

I would suggest removing the setter, and having the property react to `selectedPrinterId` which you can set whenever you need to

`this.set('selectedPrinterDisplay', { ... })` This is dangerous, it defeats the whole purpose of the property computation 🔴 

`this.set('selectedPrinterId', 123)` this doesn't enforce anyone setting it to know that the format of the object is {name,id}, once set the computed property can react to it 💚